packages/compass-shell/src/components/compass-shell/tab-compass-shell.tsx

```diff
@@ -93,26 +85,55 @@ const Shell = React.forwardRef<ShellType, ShellProps>(function Shell(
       shellRef.current?.focusEditor();
     });
   }, []);
+
+  return <_Shell ref={mergeRef} {...props}></_Shell>;
+});
+
+function useInitialEval(initialEvaluate?: string | string[]) {
+  const [initialEvalApplied, setInitialEvalApplied] = useTabState(
+    'initialEvalApplied',
+    false
+  );
+  useEffect(() => {
+    setInitialEvalApplied(true);
+  }, [setInitialEvalApplied]);
+  return initialEvalApplied ? undefined : initialEvaluate;
+}
+
+const normalizeInitialEvaluate = (initialEvaluate: string | string[]) => {
   return (
-    <_Shell
-      ref={mergeRef}
-      initialEvaluate={initialEvaluate}
-      {...props}
-    ></_Shell>
+    Array.isArray(initialEvaluate) ? initialEvaluate : [initialEvaluate]
+  ).filter((line) => {
+    // Filter out empty lines if passed by accident
+    return !!line;
+  });
+};
+
+const isInitialEvaluateEmpty = (
+  initialEvaluate?: string | string[] | undefined
+) => {
+  return (
+    !initialEvaluate || normalizeInitialEvaluate(initialEvaluate).length === 0
   );
-});
+};
 
 const CompassShell: React.FC<CompassShellProps> = ({
   runtime,
   initialHistory,
   onHistoryChange,
-  initialEvaluate,
+  initialEvaluate: _initialEvaluate,
   initialInput,
 }) => {
+  const initialEvaluate = useInitialEval(_initialEvaluate);
```

Comment:
I basically moved this from the Shell component to the CompassShell component because isOperationInProgress depends on it and we need that boolean for the shell header component too which is a peer of the shell itself.